charts/kubernetes-agent/templates/pod-clusterroles.yaml

```diff
@@ -17,12 +17,12 @@ rules:
   verbs:
   - '*'
 {{- end }}
-{{- if and .Values.persistence.nfs.watchdog.enabled .Values.persistence.nfs.enabled}}
+{{- if and .Values.persistence.nfs.watchdog.enabled (not .Values.persistence.storageClassName)}}
```

Comment:
I couldn't make this into a helper due to templating issues. We do this check in a number of places, so would be nice to make common